src/server/exchangeToken.ts

```diff
@@ -14,12 +14,15 @@ export const exchangeTokenMiddleware: Router.IMiddleware<
   WithLoggerState,
   any
 > = async (ctx: ParameterizedContext) => {
+  const logger = ctx.state.getLogger('exchangeTokenMiddleware')
+
   const nextPath = ctx.URL.searchParams.get(SearchParams.Next)
   if (!nextPath) {
     throw new Error(`Search param ${SearchParams.Next} is required`)
   }
-  // Safe way of building redirect URL.  We don't want to redirect to external site
+  // Safe way of building redirect URL. We don't want to redirect to external site
   const nextUrl = new URL('', ctx.URL)
+  nextUrl.host = process.env.HOST_URL || nextUrl.host
```

Comment:
We cannot use any VERCEL_* params for this, right?

Reply:
This is deployed on Heroku 🤔